evennia/server/serversession.py

```diff
@@ -537,3 +535,128 @@ def ndb_del(self):
     def access(self, *args, **kwargs):
         """Dummy method to mimic the logged-in API."""
         return True
+
+
+class EntitySessionHandler:
+    """
+    Handles adding/removing session references to an Account, Puppet, or perhaps
+    stranger things down the line.
+    """
+
+    def __init__(self, obj):
+        """
+        Initializes the handler.
+
+        Args:
+            obj (Object): The object on which the handler is defined.
+
+        """
+        self.obj = obj
+        self._sessions_dict = dict()
+        self._sessions = list()
```

Comment:
The original implementation fetches this from the central ServerSessionHandler. Without a change to the ServerSessionHandler, there is a risk these will go out of sync.

Reply:
They do because... I forgot to stick account.sessions.remove(self) in session.at_disconnect(). Quite forgot that Evennia doesn't normally call those methods. Now it does. I'm not having any issues on my end now. Try again?